app/views/helpers/ResultsPageHelper.scala

```diff
@@ -182,6 +197,196 @@ object ResultsPageHelper extends ViewHelper {
+           |      </a>
+           |    </li>""".stripMargin
+
+      def tabContent(marginalRate: MarginalRate) = {
+        val year = marginalRate.year
+        s"""<div class="govuk-tabs__panel" id="year$year">
+           |    <h2 class="govuk-heading-m">${messages(
+            "fullResultsPage.forFinancialYear",
+            year.toString,
+            (year + 1).toString,
+            marginalRate.days
+          )}</h2>
+           |    ${displayFullFinancialYearTable(marginalRate, associatedCompanies, taxableProfit, distributions, config)}
+           |  </div>""".stripMargin
+      }
+
+      def tabDisplay(marginalRates: Seq[MarginalRate]) =
+        Html(s"""
+                |<div class="govuk-tabs" data-module="govuk-tabs">
+                |  <h2 class="govuk-tabs__title">
+                |    Financial year results
+                |  </h2>
+                |  <ul class="govuk-tabs__list">
+                |    ${marginalRates.map(_.year).map(tabBtn).mkString}
+                |  </ul>
+                |  ${marginalRates.map(tabContent).mkString}
+                |</div>
+                |""".stripMargin)
+
+      dual.year1 -> dual.year2 match {
+        case (y1: MarginalRate, y2: MarginalRate) => tabDisplay(Seq(y1, y2))
+        case (y1: MarginalRate, y2: FlatRate)     => nonTabDisplay(Seq(y1, y2))
+        case (y1: FlatRate, y2: MarginalRate)     => nonTabDisplay(Seq(y1, y2))
+        case _                                    => throw new RuntimeException("Both financial years are flat rate")
+      }
+
+    }
+  }
+
+  private def displayFullFinancialYearTable(
+    marginalRate: MarginalRate,
+    associatedCompanies: Int,
+    taxableProfit: Long,
+    distributions: Long,
+    config:Map[Int,FYConfig]
+  )(implicit messages: Messages): Html = {
+
+    val yearConfig = config(marginalRate.year) match {
+      case x:FlatRateConfig => throw new RuntimeException("Configuration is flat where it should be marginal")
+      case x:MarginalReliefConfig => x
+    }
+
+    def boldRow(text: String) = TableRow(content = Text(text), classes = "govuk-!-font-weight-bold")
+
+    val days = marginalRate.days
+
+    val upperLimit = CurrencyUtils.format(yearConfig.upperThreshold)
+
+    val dayMsg = messages("fullResultsPage.day.singular")
+    val daysMsg = messages("fullResultsPage.day.plural")
+
+    val daysString = days match {
+      case 1 => days + s" $dayMsg"
+      case _ => days + s" $daysMsg"
+    }
+
+    val associatedCompaniesText = associatedCompanies match {
+      case 1 => 1 + " " + messages("fullResultsPage.associatedCompany.singular")
+      case x => x + " " + messages("fullResultsPage.associatedCompany.plural")
+    }
+
+    val originalCompanyMsg = messages("fullResultsPage.oneOriginalCompany")
+
+    val upperLimitMsg = messages("fullResultsPage.upperLimit")
+
+    val pointOneCompaniesCalcText = s"($associatedCompaniesText + $originalCompanyMsg)"
+
+    def cur = CurrencyUtils.format _
```

Comment:
Using CurrencyUtils.format as it is or renaming format to formatCurrency will avoid the need for this method